RSSReader/Presenter/SearchPresenter.m

```diff
@@ -0,0 +1,100 @@
+//
+//  SearchPresenter.m
+//  RSSReader
+//
+//  Created by Uladzislau Stankevich on 25.02.21.
+//
+
+#import "SearchPresenter.h"
+#import "HTMLParser.h"
+#import "RSSParser.h"
+
+@interface SearchPresenter ()
+
+@property (nonatomic, retain) NetworkManager *networkManager;
+
+@end
+
+NSString * const kNoSuchSiteMessage = @"No such site";
+NSString * const kNoRSSFeedsMessage = @"No RSS feeds";
+NSString * const kBadFormatMessage = @"Bad format";
+
+@implementation SearchPresenter
+
+- (id)initWithNetworkManager:(NetworkManager *)manager {
+    if (self = [super init]) {
+        _networkManager = [manager retain];
+    }
+    return self;
+}
+
+- (void)searchFeeds:(NSString *)url array:(NSMutableArray<SearchFeedItem *> *)array completion:(void (^)(void))completion {
+    [array retain];
+    url = [self configureURL:url];
+    if ([self validateUrl:url]) {
+        NSString *html = [self.networkManager downloadHTML:url];
+        
+        if ([html isEqualToString:@""]) {
+            [self.delegate showException:kNoSuchSiteMessage];
```

Comment:
Misleading name `showException:`
Actually you present alert controller with message.

Reply:
name changed